### PR TITLE
Export jrh

### PR DESCRIPTION
**Pull request by sriniketh923**
Import/Export of shield data has been implemented.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -36,6 +36,7 @@
 /rabbitmq-broker
 /redis-broker
 /s3
+/shieldp
```
Comment (jhunt) [inline .gitignore]: What is `shieldp`?

File: db/export.go [hunk 2]
```diff
@@ -0,0 +1,521 @@
...
+
+	for r.Next() {
+		v := job{}
+
+		if err = r.Scan(
+			&v.UUID, &v.TargetUUID, &v.StoreUUID, &v.TenantUUID,
+			&v.Name, &v.Summary, &v.Schedule, &v.KeepN, &v.KeepDays,
+			&v.NextRun, &v.Priority, &v.Paused, &v.FixedKey, &v.Healthy); err != nil {
+
+			panic(err)
```
[the remark this answers is not in the document]
Comment (jhunt) [reply db/export.go]: We might also get away with emiting the error objects into the encoded stream (i.e. `{"error":"..."}`) and either exiting early with some way of signalling to the caller not to continue with more exports, or skipping the problematic thing and letting the caller decide what to do with the data...

In any event, the eventual plugin for backing up shield ought to be able to detect errors in the output stream and change its exit code to `shield-pipe` accordingly.

File: .gitignore (hunk 1 above)
Comment (sriniketh923) [reply .gitignore]: This was the shield plugin binary. I'll remove it from this pr.

File: db/import.go [hunk 3]
```diff
@@ -549,6 +563,9 @@ func (db *DB) Import(in *json.Decoder, vault *vault.Client) error {
 				if err != nil {
 					return err
 				}
+			case "errors":
+				err := db.importErrors(in)
+				return err
```
Comment (jhunt) [inline db/import.go]: You can just return the call, since you're not doing anything with the temporary variable.

**Pull request merged**